app/badram.c

```diff
@@ -227,11 +228,11 @@ void badram_init(void)
     }
 }
 
-bool badram_insert(uintptr_t addr)
+bool badram_insert(testword_t page, testword_t offset, uintptr_t addr)
 {
     pattern_t pattern = {
-            .addr = addr,
-            .mask = DEFAULT_MASK
+        .addr = (page << PAGE_SHIFT) + offset,
```

Comment:
This will overflow for addresses > 4G on a 32-bit build. I think we need to change the `addr` field of `pattern_t` to be a `uint64_t`. That may require other changes...